ord_schema/message_helpers.py

```diff
@@ -431,6 +431,98 @@ def validate_reaction_smiles(reaction_smiles):
     return rdChemReactions.ReactionToSmiles(reaction)
 
 
+def get_compound_identifier(compound, identifier_type):
+    """Returns the value of a compound identifier if it exists.
+
+    Args:
+        compound: Compound message.
+        identifier_type: The CompoundIdentifier type to retrieve the value of.
+
+    Returns:
+        Identifier value or None if the identifier is not defined.
+    """
+    for identifier in compound.identifiers:
+        if identifier.type == identifier_type:
+            return identifier.value
+    return None
+
+
+def set_compound_identifier(compound, identifier_type, value):
+    """Sets the value of a compound identifier if it exists or creates one.
+
+    Args:
+        compound: Compound message.
+        identifier_type: The CompoundIdentifier type to retrieve the value of.
+        value: The value to set.
+
+    Returns:
+        The compound identifier that was modified or created.
+    """
+    for identifier in compound.identifiers:
+        if identifier.type == identifier_type:
+            identifier.value = value
+            return identifier
+    identifier = compound.identifiers.add(type=identifier_type, value=value)
+    return identifier
+
+
+def get_compound_smiles(compound):
```

Comment:
re: merging code

I think it would be cool if smiles_from_compound called this function in its search for existing identifiers. But it's also not super complicated so I'm not saying that's a must.

After https://github.com/Open-Reaction-Database/ord-editor/pull/11 is merged you won't need to worry about breaking the editor.